jtuples/src/main/java/org/jtuples/Tuples.java

```diff
@@ -0,0 +1,219 @@
+     * @param <B> the type of the second element of the quadruple
+     * @param <C> the type of the third element of the quadruple
+     * @param <D> the type of the fourth element of the quadruple
+     * @param first the first element of the quadruple
+     * @param second the second element of the quadruple
+     * @param third the third element of the quadruple
+     * @param fourth the fourth element of the quadruple
+     * @return a quadruple with the given arguments
+     */
+    public static <A, B, C, D> Quadruple<A, B, C, D> with(A first, B second,
+            C third, D fourth) {
+        return new Quadruple<>(first, second, third, fourth);
+    }
+    
+    /**
+     * Creates a quintuple with the given arguments.
+     * @param <A> the type of the first element of the quintuple
+     * @param <B> the type of the second element of the quintuple
+     * @param <C> the type of the third element of the quintuple
+     * @param <D> the type of the fourth element of the quintuple
+     * @param <E> the type of the fifth element of the quintuple
+     * @param first the first element of the quintuple
+     * @param second the second element of the quintuple
+     * @param third the third element of the quintuple
+     * @param fourth the fourth element of the quintuple
+     * @param fifth the fifth element of the quintuple
+     * @return a quintuple with the given arguments
+     */
+    public static <A, B, C, D, E> Quintuple<A, B, C, D, E> with(A first,
+            B second, C third, D fourth, E fifth) {
+        return new Quintuple<>(first, second, third, fourth, fifth);
+    }
+    
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatFirst(Pair<Pair<A, B>, C> pair) {
+        return new Triple<>(
+                pair.first().first(),
+                pair.first().second(),
+                pair.second()
+        );
+    }
+    
+    public static <A, B, C>
+            Triple<A, B, C> flatSecond(Pair<A, Pair<B, C>> pair) {
+        return new Triple<>(
+                pair.first(),
+                pair.second().first(),
+                pair.second().second()
+        );
+    }
+    
+    
+    public static <A, B, C>
+            Pair<Pair<A, B>, C> associateFirst(Pair<A, Pair<B, C>> pair) {
+        return new Pair<>(
+                new Pair<>(pair.first(), pair.second().first()),
+                pair.second().second()
+        );
+    }
+    
+    public static <A, B, C>
+            Pair<A, Pair<B, C>> associateSecond(Pair<Pair<A, B>, C> pair) {
+        return new Pair<>(
+                pair.first().first(),
+                new Pair<>(pair.first().second(), pair.second())
+        );
+    }
+    
+    
+    public static <A, B> Pair<B, B> map(Pair<? extends A, ? extends A> pair,
+            Function<A, B> f) {
+        return new Pair<>(
+                f.apply(pair.first()),
+                f.apply(pair.second())
+        );
+    }
```

Comment:
/**
     * Maps a function over the given tuple, resulting in a new tuple containing the results
     * of applying the function to each tuple element, in order.
     * That is, for a function {@code f} and a pair {@code (a, b)} the result is the pair {@code (f(a), f(b))}.
     */